### ISSUE TITLE
dtplyr breaks existing code using select_if, mutate_if, etc.

### ISSUE THREAD

**Issue by lutzgruber** [tidyverse/dtplyr#258]
Hi, loading `dtplyr` breaks existing code that uses the `select_if` (or `mutate_if` etc.) functions. 

```r
library(dplyr)
library(dtplyr)

data.table::data.table(
  x = c("A", "B", "C"),
  y = c("a", "b", "c")
) %>%
  mutate(z = paste0(x, y)) %>%
  select_if(is.character)
# Error: `var` must evaluate to a single number.
```

It works without `dtplyr`:

```r
library(dplyr)
# library(dtplyr)

data.table::data.table(
  x = c("A", "B", "C"),
  y = c("a", "b", "c")
) %>%
  mutate(z = paste0(x, y)) %>%
  select_if(is.character)

#         x      y      z
#    <char> <char> <char>
# 1:      A      a     Aa
# 2:      B      b     Bb
# 3:      C      c     Cc
```

I would appreciate it if this could be addressed. Thank you!

**Comment by markfairbanks** [comment]
Closing as this was fixed by #264 

``` r
library(dplyr)
library(dtplyr)

data.table::data.table(
  x = c("A", "B", "C"),
  y = c("a", "b", "c")
) %>%
  mutate(z = paste0(x, y)) %>%
  select_if(is.character)
#> Source: local data table [3 x 3]
#> Call:   copy(`_DT1`)[, `:=`(z = paste0(x, y))]
#> 
#>   x     y     z    
#>   <chr> <chr> <chr>
#> 1 A     a     Aa   
#> 2 B     b     Bb   
#> 3 C     c     Cc   
#> 
#> # Use as.data.table()/as.data.frame()/as_tibble() to access results
```

**Comment by lutzgruber-quantco** [comment]
Thank you, much appreciated!
